File: flight-status-ui/src/app/models/auth.models.ts
export interface LoginRequest   { email: string; password: string; }
export interface RegisterRequest { firstName: string; lastName: string; email: string; password: string; }
export interface LoginResponse  {
  token: string; email: string; firstName: string; lastName: string;
  role: string; expiresAt: string;
}
export interface CurrentUser { email: string; firstName: string; lastName: string; role: string; }

File: FlightStatus.Api/Models/Auth/AuthModels.cs
namespace FlightStatus.Api.Models.Auth;

public sealed record RegisterRequest
{
    public required string FirstName { get; init; }
    public required string LastName { get; init; }
    public required string Email { get; init; }
    public required string Password { get; init; }
}

public sealed record LoginRequest
{
    public required string Email { get; init; }
    public required string Password { get; init; }
}

public sealed record LoginResponse
{
    public required string Token { get; init; }
    public required string Email { get; init; }
    public required string FirstName { get; init; }
    public required string LastName { get; init; }
    public required string Role { get; init; }
    public DateTime ExpiresAt { get; init; }
}
